File: snapbuy-server/src/main/java/com/g127/snapbuy/service/SupplierReportService.java
package com.g127.snapbuy.service;

import com.g127.snapbuy.dto.response.SupplierReportResponse;

import java.time.LocalDate;

public interface SupplierReportService {
    SupplierReportResponse getDailySupplierReport(LocalDate date);
    SupplierReportResponse getMonthlySupplierReport(int year, int month);
    SupplierReportResponse getYearlySupplierReport(int year);
    SupplierReportResponse getCustomSupplierReport(LocalDate startDate, LocalDate endDate);
}

File: snapbuy-client/src/services/SupplierReportService.js
import axios from "axios";
import { API_ENDPOINTS, getAuthHeaders } from "./apiConfig";

const REST_API_BASE_URL = API_ENDPOINTS.REPORTS.SUPPLIERS;

// Lấy báo cáo nhà cung cấp theo ngày
export const getDailySupplierReport = async (date) => {
  const response = await axios.get(
    `${REST_API_BASE_URL}/daily?date=${date}`,
    getAuthHeaders()
  );
  return response.data?.result || response.data;
};

// Lấy báo cáo nhà cung cấp theo tháng
export const getMonthlySupplierReport = async (year, month) => {
  const response = await axios.get(
    `${REST_API_BASE_URL}/monthly?year=${year}&month=${month}`,
    getAuthHeaders()
  );
  return response.data?.result || response.data;
};

// Lấy báo cáo nhà cung cấp theo năm
export const getYearlySupplierReport = async (year) => {
  const response = await axios.get(
    `${REST_API_BASE_URL}/yearly?year=${year}`,
    getAuthHeaders()
  );
  return response.data?.result || response.data;
};

// Lấy báo cáo nhà cung cấp theo khoảng thời gian tùy chỉnh
export const getCustomSupplierReport = async (startDate, endDate) => {
  const response = await axios.get(
    `${REST_API_BASE_URL}/custom?startDate=${startDate}&endDate=${endDate}`,
    getAuthHeaders()
  );
  return response.data?.result || response.data;
};
